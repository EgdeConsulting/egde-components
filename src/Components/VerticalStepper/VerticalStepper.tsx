/* eslint-disable */
import React, { ReactElement } from 'react';
import {
    Menu,
    MenuButton,
    MenuList,
    MenuItem,
    Button,
    Flex,
    Heading,
    Center,
    Text,
} from '@chakra-ui/react';
import { ChevronDownIcon } from '@chakra-ui/icons';
import { StepperProps } from 'Types';

const EditIcon = (): ReactElement => {
    return (
        <svg
            width='40'
            height='40'
            viewBox='0 0 40 40'
            fill='none'
            xmlns='http://www.w3.org/2000/svg'
        >
            <path
                d='M0 20C0 8.95431 8.95431 0 20 0C31.0457 0 40 8.95431 40 20C40 31.0457 31.0457 40 20 40C8.95431 40 0 31.0457 0 20Z'
                fill='#027A1F'
            />
            <path
                d='M22.06 17L23 17.94L13.92 27H13V26.08L22.06 17ZM25.66 11C25.41 11 25.15 11.1 24.96 11.29L23.13 13.12L26.88 16.87L28.71 15.04C29.1 14.65 29.1 14 28.71 13.63L26.37 11.29C26.17 11.09 25.92 11 25.66 11ZM22.06 14.19L11 25.25V29H14.75L25.81 17.94L22.06 14.19Z'
                fill='white'
            />
        </svg>
    );
};

const CheckIcon = (): ReactElement => {
    return (
        <svg
            width='41'
            height='40'
            viewBox='0 0 41 40'
            fill='none'
            xmlns='http://www.w3.org/2000/svg'
        >
            <path
                d='M0.910156 20C0.910156 8.95431 9.86446 0 20.9102 0C31.9559 0 40.9102 8.95431 40.9102 20C40.9102 31.0457 31.9559 40 20.9102 40C9.86446 40 0.910156 31.0457 0.910156 20Z'
                fill='#027A1F'
            />
            <path
                d='M18.9109 23.1715L28.1029 13.9785L29.5179 15.3925L18.9109 25.9995L12.5469 19.6355L13.9609 18.2215L18.9109 23.1715Z'
                fill='white'
            />
        </svg>
    );
};

export const VerticalStepper = (props: StepperProps): ReactElement => {
    const {
        activeStep,
        setActiveStep,
        stepsContent,
        buttonTexts,
        finalStep,
        completedSteps,
        onFinalize,
        onFinalStep,
        isCompleteButtonDisabled,
        buttonCaption,
    } = props;
    const { previous, next, complete } = buttonTexts;
    const { finalText, finalButtonText } = finalStep;
    const activeStepContent = stepsContent.find(
        (_, index) => index === activeStep
    );
    const isStepCompleted = completedSteps?.includes(activeStep);
    return (
        <Flex flexDir='column' width='100%'>
            <Menu flip={false} matchWidth={true} offset={[0, 0]}>
                <MenuButton
                    backgroundColor='white'
                    padding='8px'
                    borderWidth='1px'
                    borderColor='border'
                    shadow='base'
                >
                    <Flex width='100%'>
                        {isStepCompleted ? <CheckIcon /> : <EditIcon />}
                        <Flex width='100%' justifyContent='space-between'>
                            <Flex
                                direction='column'
                                alignItems='start'
                                marginLeft='1rem'
                            >
                                <Heading size='md' textAlign='left'>
                                    {activeStep + 1} av {stepsContent.length}{' '}
                                    {stepsContent[activeStep].label}
                                </Heading>
                                {activeStep < stepsContent.length - 1 && (
                                    <Text textAlign='left' marginBottom='0'>
                                        Neste:{' '}
                                        {stepsContent[activeStep + 1].label}
                                    </Text>
                                )}
                            </Flex>
                            <ChevronDownIcon
                                color='blue'
                                marginLeft='0.5rem'
                                fontSize='1.5rem'
                            />
                        </Flex>
                    </Flex>
                </MenuButton>
                <MenuList>
                    {stepsContent.map(({ label }, index) => (
                        <MenuItem>
                            {label === activeStepContent?.label ? (
                                <Text marginBottom='0'>
                                    {index + 1} av {stepsContent.length} {label}
                                </Text>
                            ) : (
                                <Button
                                    variant='link'
                                    fontWeight='normal'
                                    onClick={() => setActiveStep(index)}
                                >
                                    {index + 1} av {stepsContent.length} {label}
                                </Button>
                            )}
                        </MenuItem>
                    ))}
                </MenuList>
            </Menu>
            {stepsContent[activeStep].children}
            {activeStep === stepsContent.length ? (
                <Center flexDir='column'>
                    <Heading>{finalText}</Heading>
                    <Button
                        variant='primary'
                        marginStart='0.75rem'
                        onClick={() => onFinalize()}
                    >
                        {finalButtonText}
                    </Button>
                </Center>
            ) : (
                <Flex width='100%' direction='column'>
                    {activeStep !== 0 && (
                        <Button
                            variant='secondary'
                            marginStart='0.75rem'
                            marginBottom='1rem'
                            onClick={() => {
                                setActiveStep((prev) => prev - 1);
                                window.scrollTo(0, 227);
                            }}
                        >
                            {previous}
                        </Button>
                    )}

                    <Button
                        isDisabled={
                            isCompleteButtonDisabled
                            && activeStep === stepsContent.length - 1
                        }
                        variant='primary'
                        marginStart='0.75rem'
                        onClick={
                            activeStep === stepsContent.length - 1
                                ? () => {
                                      onFinalStep();
                                      window.scrollTo(0, 227);
                                  }
                                : () => {
                                      setActiveStep((prev) => prev + 1);
                                      window.scrollTo(0, 227);
                                  }
                        }
                    >
                        {activeStep === stepsContent.length - 1
                            ? complete
                            : next}
                    </Button>
                    {buttonCaption && (
                        <Text color={buttonCaption.color}>
                            {buttonCaption.text}
                        </Text>
                    )}
                </Flex>
            )}
        </Flex>
    );
};
