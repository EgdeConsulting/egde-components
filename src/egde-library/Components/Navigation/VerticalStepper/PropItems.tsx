import React from 'react';

export const Default = [
    {
        label: 'Step 1',
        children: <div>Content of Step 1</div>,
    },
    {
        label: 'Step 2',
        children: <div>Content of Step 2</div>,
    },
    {
        label: 'Step 3',
        children: <div>Content of Step 3</div>,
    },
    {
        label: 'Step 4',
        children: <div>Content of Step 4</div>,
    },
];

export const ManySteps = [
    {
        label: 'Step 1',
        children: <div> Content of Step 1</div>,
    },
    {
        label: 'Step 2',
        children: <div>Content of Step 2</div>,
    },
    {
        label: 'Step 3',
        children: <div>Content of Step 3</div>,
    },
    {
        label: 'Step 4',
        children: <div>Content of Step 4</div>,
    },
    {
        label: 'Step 5',
        children: <div>Content of Step 5</div>,
    },
    {
        label: 'Step 6',
        children: <div>Content of Step 6</div>,
    },
    {
        label: 'Step 7',
        children: <div>Content of Step 7</div>,
    },
    {
        label: 'Step 8',
        children: <div>Content of Step 8</div>,
    },
    {
        label: 'Step 9',
        children: <div>Content of Step 9</div>,
    },
    {
        label: 'Step 10',
        children: <div>Content of Step 10</div>,
    },
];
