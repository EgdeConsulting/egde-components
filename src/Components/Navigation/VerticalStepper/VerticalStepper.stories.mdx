import { Meta, Story, Canvas } from '@storybook/addon-docs';
import { DocsPageHeader, DocsPageBodyComponents } from 'Docs';
import { LibPath } from 'Types';
import { VerticalStepper } from 'Components';
import * as PropItems from './PropItems';
import { useState } from 'react';
import { Button } from '@chakra-ui/react';

export const Template = (args) => {
    const [activeStep, setActiveStep] = useState(0);
    return (
        <div>
            <VerticalStepper
                {...args}
                setActiveStep={setActiveStep}
                activeStep={activeStep}
            />
        </div>
    );
};

export const componentName = 'VerticalStepper';

<Meta
    title={LibPath.Navigation + componentName}
    component={VerticalStepper}
    args={{
        activeStep: 0,
        setActiveStep: () => {},
        stepsContent: PropItems.Default,
        buttonTexts: {
            previous: 'Previous',
            next: 'Next',
            complete: 'Complete',
        },
        finalStep: {
            finalText: 'Final Step',
            finalButtonText: 'Finish',
        },
        completedSteps: [],
        isCompleteButtonDisabled: false,
        buttonCaption: {
            text: 'Caption:',
            color: '#000000',
        },
    }}
      argtypes={{
        onFinalize: { action: 'Finalized' },
        onFinalStep: { action: 'FinalStep' },
    }}
/>

<DocsPageHeader
    title={componentName}
    description="This component is used as navigation between steps."
    addImportantNote={false}
    variant="stable"
/>

<Canvas withToolbar>
    <Story
        name="Base"
        height="300px"
        args={{
            isCompleteButtonDisabled: true,
        }}
    >
        {Template.bind({})}
    </Story>
</Canvas>

<DocsPageBodyComponents addStories={true} />

### Completed Steps

<Canvas withToolbar>
    <Story
        name="Completed Steps"
        height="300px"
        args={{
            activeStep: 4,
            completedSteps: [0, 1, 2, 3, 4],
            isCompleteButtonDisabled: false,
        }}
    >
        {Template.bind({})}
    </Story>
</Canvas>

### Many Steps

<Canvas withToolbar>
    <Story
        name="Many Steps"
        height="430px"
        args={{
            stepsContent: PropItems.ManySteps,
            isCompleteButtonDisabled: true,
        }}
    >
        {Template.bind({})}
    </Story>
</Canvas>
