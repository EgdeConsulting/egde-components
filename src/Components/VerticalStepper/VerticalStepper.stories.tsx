import React, { useState } from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';
import { VerticalStepper } from './VerticalStepper';
import { LibPath } from 'Types';

export default {
  title: LibPath.Navigation + 'VerticalStepper' || 'undefined',
  component: VerticalStepper,
  argTypes: {
    stepsContent: {
      defaultValue: [
        {
          label: 'Step 1',
          children: <div>Step 1</div>,
        },
        {
          label: 'Step 2',
          children: <div>Step 2</div>,
        },
        {
          label: 'Step 3',
          children: <div>Step 3</div>,
        },
        {
          label: 'Step 4',
          children: <div>Step 4</div>,
        },
        {
          label: 'Step 5',
          children: <div>Step 5</div>,
        },
      ],
    },
    buttonTexts: {
      defaultValue:
        {
          previous: 'Previous',
          next: 'Next',
          complete: 'Complete',
        },
    },
    finalStep: {
      defaultValue: {
        finalText: 'Final Step',
        finalButtonText: 'Finish',
      },
    },
    completedSteps: { defaultValue: [] },
    onFinalize: { action: 'onFinalize' },
    onFinalStep: { action: 'onClose' },
    isCompleteButtonDisabled: { defaultValue: false },
    buttonCaption: {
      defaultValue: {
        text: 'Caption:',
        color: '#000000',
      },
    },
  },
} as ComponentMeta<typeof VerticalStepper>;

const Template: ComponentStory<typeof VerticalStepper> = (args) => {
    const [activeStep, setActiveStep] = useState(0);
    return (
        <div>
             <VerticalStepper {...args} setActiveStep={setActiveStep} activeStep={activeStep} />
        </div>
    )
   
}


export const VerticalStepperTest = Template.bind({});
