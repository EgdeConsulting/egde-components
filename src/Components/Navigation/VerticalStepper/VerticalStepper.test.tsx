import React from 'react';
import { render } from '@testing-library/react';
import { VerticalStepper } from './VerticalStepper';

// Snapshot Test
describe('Snapshot VerticalStepper', () => {
  it('should render correctly', () => {
    const tree = render(<VerticalStepper
        activeStep= {0}
        setActiveStep= {() => {}}
        stepsContent= {[
            {
                label: 'Step 1',
                children: <div> Content of Step 1</div>,
            },
            {
                label: 'Step 2',
                children: <div>Content of Step 2</div>,
            },
            {
                label: 'Step 3',
                children: <div>Content of Step 3</div>,
            },
            {
                label: 'Step 4',
                children: <div>Content of Step 4</div>,
            },
            {
                label: 'Step 5',
                children: <div>Content of Step 5</div>,
            },
        ]}
        buttonTexts= {{
            previous: 'Previous',
            next: 'Next',
            complete: 'Complete',
        }}

        finalStep={{
            finalText: 'Final Step',
            finalButtonText: 'Finish',
        }}
        completedSteps= {[]}
        onFinalize= {() => {}}
        onFinalStep= {() => {}}
        isCompleteButtonDisabled= {false}
        buttonCaption= {{
            text: 'Caption:',
            color: '#000000',
        }}
        />
        );
    expect(tree).toMatchSnapshot();
  });
});


