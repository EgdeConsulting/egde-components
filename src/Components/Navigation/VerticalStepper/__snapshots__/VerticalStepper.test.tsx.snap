// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClickableVerticalStepper component test matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class="chakra-ui-light"
  >
    <div
      id="chakra-toast-portal"
    >
      <ul
        id="chakra-toast-manager-top"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-top-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-top-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-bottom-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
    </div>
    <div>
      <div
        class="css-1ofqig9"
      >
        <button
          aria-controls="menu-list-5"
          aria-expanded="false"
          aria-haspopup="menu"
          class="chakra-menu__menu-button css-b2lpn5"
          id="menu-button-5"
        >
          <span
            class="css-xl71ch"
          >
            <div
              class="css-hboir5"
            >
              <svg
                fill="none"
                height="40"
                viewBox="0 0 40 40"
                width="40"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 20C0 8.95431 8.95431 0 20 0C31.0457 0 40 8.95431 40 20C40 31.0457 31.0457 40 20 40C8.95431 40 0 31.0457 0 20Z"
                  fill="#027A1F"
                />
                <path
                  d="M22.06 17L23 17.94L13.92 27H13V26.08L22.06 17ZM25.66 11C25.41 11 25.15 11.1 24.96 11.29L23.13 13.12L26.88 16.87L28.71 15.04C29.1 14.65 29.1 14 28.71 13.63L26.37 11.29C26.17 11.09 25.92 11 25.66 11ZM22.06 14.19L11 25.25V29H14.75L25.81 17.94L22.06 14.19Z"
                  fill="white"
                />
              </svg>
              <div
                class="css-z7mtfw"
              >
                <div
                  class="css-y0fxms"
                >
                  <h2
                    class="chakra-heading css-1qeza09"
                  >
                    1
                     av 
                    3
                     
                    first step
                  </h2>
                  <p
                    class="chakra-text css-1g2fpq9"
                  >
                    Neste:
                     
                    second step
                  </p>
                </div>
                <svg
                  class="chakra-icon css-1puyyru"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                    fill="currentColor"
                  />
                </svg>
              </div>
            </div>
          </span>
        </button>
        <div
          class="css-r6z5ec"
          style="visibility: hidden; position: absolute; min-width: max-content; inset: 0 auto auto 0;"
        >
          <div
            aria-orientation="vertical"
            class="chakra-menu__menu-list css-1ozmk1d"
            id="menu-list-5"
            role="menu"
            style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.8) translateZ(0);"
            tabindex="-1"
          >
            <button
              class="chakra-menu__menuitem css-13c7rae"
              data-index="0"
              id="menu-list-5-menuitem-1"
              role="menuitem"
              tabindex="-1"
              type="button"
            >
              <p
                class="chakra-text css-7bug7u"
              >
                1
                 av 
                3
                 
                first step
              </p>
            </button>
            <button
              class="chakra-menu__menuitem css-13c7rae"
              data-index="1"
              id="menu-list-5-menuitem-2"
              role="menuitem"
              tabindex="-1"
              type="button"
            >
              <a
                class="chakra-link css-1kynmnd"
              >
                2
                 av 
                3
                 
                second step
              </a>
            </button>
            <button
              class="chakra-menu__menuitem css-13c7rae"
              data-index="2"
              id="menu-list-5-menuitem-3"
              role="menuitem"
              tabindex="-1"
              type="button"
            >
              <a
                class="chakra-link css-1kynmnd"
              >
                3
                 av 
                3
                 
                third step
              </a>
            </button>
          </div>
        </div>
        text of first step
        <div
          class="css-1ofqig9"
        >
          <button
            class="chakra-button css-1kg50ao"
            type="button"
          >
            next
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-1ofqig9"
    >
      <button
        aria-controls="menu-list-5"
        aria-expanded="false"
        aria-haspopup="menu"
        class="chakra-menu__menu-button css-b2lpn5"
        id="menu-button-5"
      >
        <span
          class="css-xl71ch"
        >
          <div
            class="css-hboir5"
          >
            <svg
              fill="none"
              height="40"
              viewBox="0 0 40 40"
              width="40"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 20C0 8.95431 8.95431 0 20 0C31.0457 0 40 8.95431 40 20C40 31.0457 31.0457 40 20 40C8.95431 40 0 31.0457 0 20Z"
                fill="#027A1F"
              />
              <path
                d="M22.06 17L23 17.94L13.92 27H13V26.08L22.06 17ZM25.66 11C25.41 11 25.15 11.1 24.96 11.29L23.13 13.12L26.88 16.87L28.71 15.04C29.1 14.65 29.1 14 28.71 13.63L26.37 11.29C26.17 11.09 25.92 11 25.66 11ZM22.06 14.19L11 25.25V29H14.75L25.81 17.94L22.06 14.19Z"
                fill="white"
              />
            </svg>
            <div
              class="css-z7mtfw"
            >
              <div
                class="css-y0fxms"
              >
                <h2
                  class="chakra-heading css-1qeza09"
                >
                  1
                   av 
                  3
                   
                  first step
                </h2>
                <p
                  class="chakra-text css-1g2fpq9"
                >
                  Neste:
                   
                  second step
                </p>
              </div>
              <svg
                class="chakra-icon css-1puyyru"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </div>
        </span>
      </button>
      <div
        class="css-r6z5ec"
        style="visibility: hidden; position: absolute; min-width: max-content; inset: 0 auto auto 0;"
      >
        <div
          aria-orientation="vertical"
          class="chakra-menu__menu-list css-1ozmk1d"
          id="menu-list-5"
          role="menu"
          style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.8) translateZ(0);"
          tabindex="-1"
        >
          <button
            class="chakra-menu__menuitem css-13c7rae"
            data-index="0"
            id="menu-list-5-menuitem-1"
            role="menuitem"
            tabindex="-1"
            type="button"
          >
            <p
              class="chakra-text css-7bug7u"
            >
              1
               av 
              3
               
              first step
            </p>
          </button>
          <button
            class="chakra-menu__menuitem css-13c7rae"
            data-index="1"
            id="menu-list-5-menuitem-2"
            role="menuitem"
            tabindex="-1"
            type="button"
          >
            <a
              class="chakra-link css-1kynmnd"
            >
              2
               av 
              3
               
              second step
            </a>
          </button>
          <button
            class="chakra-menu__menuitem css-13c7rae"
            data-index="2"
            id="menu-list-5-menuitem-3"
            role="menuitem"
            tabindex="-1"
            type="button"
          >
            <a
              class="chakra-link css-1kynmnd"
            >
              3
               av 
              3
               
              third step
            </a>
          </button>
        </div>
      </div>
      text of first step
      <div
        class="css-1ofqig9"
      >
        <button
          class="chakra-button css-1kg50ao"
          type="button"
        >
          next
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`UnclickableVerticalStepper component test matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class="chakra-ui-light"
  >
    <div
      id="chakra-toast-portal"
    >
      <ul
        id="chakra-toast-manager-top"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-top-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-top-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-bottom-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
    </div>
    <div>
      <div
        class="css-1ofqig9"
      >
        <button
          aria-controls="menu-list-5"
          aria-expanded="false"
          aria-haspopup="menu"
          class="chakra-menu__menu-button css-b2lpn5"
          id="menu-button-5"
        >
          <span
            class="css-xl71ch"
          >
            <div
              class="css-hboir5"
            >
              <svg
                fill="none"
                height="40"
                viewBox="0 0 40 40"
                width="40"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 20C0 8.95431 8.95431 0 20 0C31.0457 0 40 8.95431 40 20C40 31.0457 31.0457 40 20 40C8.95431 40 0 31.0457 0 20Z"
                  fill="#027A1F"
                />
                <path
                  d="M22.06 17L23 17.94L13.92 27H13V26.08L22.06 17ZM25.66 11C25.41 11 25.15 11.1 24.96 11.29L23.13 13.12L26.88 16.87L28.71 15.04C29.1 14.65 29.1 14 28.71 13.63L26.37 11.29C26.17 11.09 25.92 11 25.66 11ZM22.06 14.19L11 25.25V29H14.75L25.81 17.94L22.06 14.19Z"
                  fill="white"
                />
              </svg>
              <div
                class="css-z7mtfw"
              >
                <div
                  class="css-y0fxms"
                >
                  <h2
                    class="chakra-heading css-1qeza09"
                  >
                    1
                     av 
                    3
                     
                    first step
                  </h2>
                  <p
                    class="chakra-text css-1g2fpq9"
                  >
                    Neste:
                     
                    second step
                  </p>
                </div>
                <svg
                  class="chakra-icon css-1puyyru"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                    fill="currentColor"
                  />
                </svg>
              </div>
            </div>
          </span>
        </button>
        <div
          class="css-r6z5ec"
          style="visibility: hidden; position: absolute; min-width: max-content; inset: 0 auto auto 0;"
        >
          <div
            aria-orientation="vertical"
            class="chakra-menu__menu-list css-1ozmk1d"
            id="menu-list-5"
            role="menu"
            style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.8) translateZ(0);"
            tabindex="-1"
          >
            <button
              class="chakra-menu__menuitem css-13c7rae"
              data-index="0"
              id="menu-list-5-menuitem-1"
              role="menuitem"
              tabindex="-1"
              type="button"
            >
              <p
                class="chakra-text css-7bug7u"
              >
                1
                 av 
                3
                 
                first step
              </p>
            </button>
            <button
              class="chakra-menu__menuitem css-13c7rae"
              data-index="1"
              id="menu-list-5-menuitem-2"
              role="menuitem"
              tabindex="-1"
              type="button"
            >
              <a
                class="chakra-link css-1kynmnd"
              >
                2
                 av 
                3
                 
                second step
              </a>
            </button>
            <button
              class="chakra-menu__menuitem css-13c7rae"
              data-index="2"
              id="menu-list-5-menuitem-3"
              role="menuitem"
              tabindex="-1"
              type="button"
            >
              <a
                class="chakra-link css-1kynmnd"
              >
                3
                 av 
                3
                 
                third step
              </a>
            </button>
          </div>
        </div>
        text of first step
        <div
          class="css-1ofqig9"
        >
          <button
            class="chakra-button css-1kg50ao"
            type="button"
          >
            next
          </button>
          <p
            class="chakra-text css-o8mki"
          >
            ButtonCaption
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-1ofqig9"
    >
      <button
        aria-controls="menu-list-5"
        aria-expanded="false"
        aria-haspopup="menu"
        class="chakra-menu__menu-button css-b2lpn5"
        id="menu-button-5"
      >
        <span
          class="css-xl71ch"
        >
          <div
            class="css-hboir5"
          >
            <svg
              fill="none"
              height="40"
              viewBox="0 0 40 40"
              width="40"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 20C0 8.95431 8.95431 0 20 0C31.0457 0 40 8.95431 40 20C40 31.0457 31.0457 40 20 40C8.95431 40 0 31.0457 0 20Z"
                fill="#027A1F"
              />
              <path
                d="M22.06 17L23 17.94L13.92 27H13V26.08L22.06 17ZM25.66 11C25.41 11 25.15 11.1 24.96 11.29L23.13 13.12L26.88 16.87L28.71 15.04C29.1 14.65 29.1 14 28.71 13.63L26.37 11.29C26.17 11.09 25.92 11 25.66 11ZM22.06 14.19L11 25.25V29H14.75L25.81 17.94L22.06 14.19Z"
                fill="white"
              />
            </svg>
            <div
              class="css-z7mtfw"
            >
              <div
                class="css-y0fxms"
              >
                <h2
                  class="chakra-heading css-1qeza09"
                >
                  1
                   av 
                  3
                   
                  first step
                </h2>
                <p
                  class="chakra-text css-1g2fpq9"
                >
                  Neste:
                   
                  second step
                </p>
              </div>
              <svg
                class="chakra-icon css-1puyyru"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </div>
        </span>
      </button>
      <div
        class="css-r6z5ec"
        style="visibility: hidden; position: absolute; min-width: max-content; inset: 0 auto auto 0;"
      >
        <div
          aria-orientation="vertical"
          class="chakra-menu__menu-list css-1ozmk1d"
          id="menu-list-5"
          role="menu"
          style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.8) translateZ(0);"
          tabindex="-1"
        >
          <button
            class="chakra-menu__menuitem css-13c7rae"
            data-index="0"
            id="menu-list-5-menuitem-1"
            role="menuitem"
            tabindex="-1"
            type="button"
          >
            <p
              class="chakra-text css-7bug7u"
            >
              1
               av 
              3
               
              first step
            </p>
          </button>
          <button
            class="chakra-menu__menuitem css-13c7rae"
            data-index="1"
            id="menu-list-5-menuitem-2"
            role="menuitem"
            tabindex="-1"
            type="button"
          >
            <a
              class="chakra-link css-1kynmnd"
            >
              2
               av 
              3
               
              second step
            </a>
          </button>
          <button
            class="chakra-menu__menuitem css-13c7rae"
            data-index="2"
            id="menu-list-5-menuitem-3"
            role="menuitem"
            tabindex="-1"
            type="button"
          >
            <a
              class="chakra-link css-1kynmnd"
            >
              3
               av 
              3
               
              third step
            </a>
          </button>
        </div>
      </div>
      text of first step
      <div
        class="css-1ofqig9"
      >
        <button
          class="chakra-button css-1kg50ao"
          type="button"
        >
          next
        </button>
        <p
          class="chakra-text css-o8mki"
        >
          ButtonCaption
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
